### PR TITLE
Request validation should not throw if verifyingContract is not defined in typed signature

### PR DESCRIPTION
**Pull request by jpuri**
https://github.com/MetaMask/metamask-extension/issues/26908

### CODE REVIEW

File: src/wallet.ts [hunk 1]
```diff
@@ -463,10 +463,8 @@ WalletMiddlewareOptions): JsonRpcMiddleware<any, Block> {
  * @param data - The data passed in typedSign request.
  */
 function validateVerifyingContract(data: string) {
-  const {
-    domain: { verifyingContract },
-  } = parseTypedMessage(data);
-  if (!isValidHexAddress(verifyingContract)) {
+  const { domain: { verifyingContract } = {} } = parseTypedMessage(data);
+  if (verifyingContract && !isValidHexAddress(verifyingContract)) {
```
Comment (jiexi) [inline src/wallet.ts:467]: Does it matter if `verifyingContract` is an empty string?

**Review by pedronfigueiredo — APPROVED** [review]

File: src/wallet.ts (hunk 1 above)
Comment (jpuri) [reply src/wallet.ts:467]: An empty value will not throw error.

**Review by jiexi — APPROVED** [review]

File: src/wallet.ts (hunk 1 above)
Comment (Gudahtt) [reply src/wallet.ts:467]: I think @jiexi meant to ask _should_ it throw an error.

I've just tested with v12.0.6 and it looks like an empty `verifyingContract` was accepted though, so continuing to accept it sounds reasonable for now. It did trigger a blockaid warning on v12.0.6, but no crash, and I was able to sign it.

File: src/wallet.test.ts [hunk 2]
```diff
@@ -524,6 +568,35 @@ describe('wallet', () => {
...
+        createWalletMiddleware({ getAccounts, processTypedMessageV4 }),
+      );
+
+      const payload = {
+        method: 'eth_signTypedData_v4',
+        params: [testAddresses[0], JSON.stringify(getMsgParams())],
+      };
+
+      const promise = pify(engine.handle).call(engine, payload);
+      const result = await promise;
```
Comment (Gudahtt) [inline src/wallet.test.ts:592]: Nit: Unnecessary deferred Promise:
```suggestion
      const result = await pify(engine.handle).call(engine, payload);
```

File: src/wallet.test.ts [hunk 3]
```diff
@@ -432,6 +432,50 @@ describe('wallet', () => {
       const promise = pify(engine.handle).call(engine, payload);
       await expect(promise).rejects.toThrow('Invalid input.');
     });
+
+    it('should not throw if verifyingContract is undefined', async () => {
+      const { engine } = createTestSetup();
+      const getAccounts = async () => testAddresses.slice();
+      const witnessedMsgParams: TypedMessageParams[] = [];
+      const processTypedMessageV3 = async (msgParams: TypedMessageParams) => {
+        witnessedMsgParams.push(msgParams);
```
Comment (Gudahtt) [inline src/wallet.test.ts:441]: Nit: This variable seems to serve no purpose:
```suggestion
      const processTypedMessageV3 = async (msgParams: TypedMessageParams) => {
```

File: src/wallet.test.ts [hunk 4]
```diff
@@ -524,6 +568,35 @@ describe('wallet', () => {
       const promise = pify(engine.handle).call(engine, payload);
       await expect(promise).rejects.toThrow('Invalid input.');
     });
+
+    it('should not throw if request is permit with undefined value for verifyingContract address', async () => {
+      const { engine } = createTestSetup();
+      const getAccounts = async () => testAddresses.slice();
+      const witnessedMsgParams: TypedMessageParams[] = [];
+      const processTypedMessageV4 = async (msgParams: TypedMessageParams) => {
+        witnessedMsgParams.push(msgParams);
```
Comment (Gudahtt) [inline src/wallet.test.ts:577]: Nit: This variable isn't used either
```suggestion
      const processTypedMessageV4 = async (msgParams: TypedMessageParams) => {
```

**Review by Gudahtt — APPROVED** [review]
LGTM! Some test cleanup suggested, but the tests were already pretty messy to start with

**Pull request merged**